class-two-factor-core.php

```diff
@@ -855,7 +855,7 @@ public static function login_form_validate_2fa() {
 
 		// Ask the provider to verify the second factor.
 		if ( true !== $provider->validate_authentication( $user ) ) {
-			do_action( 'wp_login_failed', $user->user_login );
+			do_action( 'wp_login_failed', $user->user_login, new WP_Error( 'two_factor_invalid', esc_html__( 'ERROR: Invalid verification code.', 'two-factor' ) ) );
```

Comment:
It appears that core doesn’t escape the error message when returning it as part of the error object:

https://github.com/WordPress/wordpress-develop/blob/6.0.2/src/wp-includes/pluggable.php#L625

I think we should also let it be escaped later, if needed by the action hook user method.

Reply:
To be honest, Core correctly doesn't escape any static string, and I believe not escaping translations is the correct thing to do.

I only escaped it here to be inline with the line below where I copied it from. I'll switch it over for consistency.